packages/@react-spectrum/menu/docs/Menu.mdx

```diff
@@ -0,0 +1,357 @@
+
+```tsx example
+function Example() {
+  let [selected, setSelected] = React.useState(new Set(['Sidebar', 'Console']));
+
+  return (
+    <>
+      <MenuTrigger closeOnSelect={false}>
+        <ActionButton>
+          Show
+        </ActionButton>
+        <Menu selectionMode="multiple" selectedKeys={selected} onSelectionChange={setSelected}>
+          <Item key='Sidebar'>Sidebar</Item>
+          <Item key='Searchbar'>Searchbar</Item>
+          <Item key='Tools'>Tools</Item>
+          <Item key='Console'>Console</Item>
+        </Menu>
+      </MenuTrigger>
+      <p>Current selection (controlled): {[...selected].join(', ')}</p>
+    </>
+  );
+}
+```
+
+## Complex Menu Items
+
+Items within Menu also allow for additional content used to better communicate options. Icons and descriptions can be added to the `children` of `Item` as shown in the example below. If a description is added, the prop `slot="description"` must be used to distinguish the different `<Text>` elements.
+
+[View guidelines](https://spectrum.adobe.com/page/popovers/#Content-types)
+
+```tsx example
+import {Keyboard, Text} from '@react-spectrum/text';
+<MenuTrigger>
+  <ActionButton>
+    Edit
+  </ActionButton>
+  <Menu>
+    <Item key="cut" textValue="cut">
+      <Cut />
+      <Text>Cut</Text>
+      <Keyboard>⌘X</Keyboard>
+    </Item>
+    <Item key="copy" textValue="copy">
+      <Copy />
+      <Text>Copy</Text>
+      <Keyboard>⌘C</Keyboard>
+    </Item>
+    <Item key="paste" textValue="paste">
+      <Paste />
+      <Text>Paste</Text>
+      <Keyboard>⌘V</Keyboard>
+    </Item>
+  </Menu>
+</MenuTrigger>
+```
+
+## Props
+
+<PropTable component={docs.exports.Menu} links={docs.links} />
+
+## Visual Options
+
+### Autofocus (Menu)
+
+Applying `autoFocus` to the Menu of the MenuTrigger sets focus to a Menu Item
+within the Menu upon opening.
+
+This example demonstrates how to use `autoFocus` to automatically focus the
+selected Menu Item when the Menu is opened.
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+    autoFocus
+  </ActionButton>
+  <Menu
+    selectionMode="single"
+    selectedKeys={['copy']}>
+    <Item key="cut">Cut</Item>
+    <Item key="copy">Copy</Item>
+    <Item key="paste">Paste</Item>
+  </Menu>
+</MenuTrigger>
+```
+
+These examples demonstrate how to use `autoFocus` to set whether or not the
+first Menu Item or last Menu Item is focused when the Menu is opened.
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+      autofocus=first
+  </ActionButton>
+  <Menu autoFocus="first">
+    <Item key="cut">Cut</Item>
+    <Item key="copy">Copy</Item>
+    <Item key="paste">Paste</Item>
+  </Menu>
+</MenuTrigger>
+<MenuTrigger>
+  <ActionButton>
+    autofocus=last
+  </ActionButton>
+  <Menu autoFocus="last">
+    <Item key="cut">Cut</Item>
+    <Item key="copy">Copy</Item>
+    <Item key="paste">Paste</Item>
+  </Menu>
+</MenuTrigger>
+```
+
+### Disabled
+
+```tsx example
```

Comment:
This example throws an error when you click the button: `No key found for item`.